### PR TITLE
Tickets/dm 47843

### PR DESCRIPTION
**Pull request by teweitsai**
* Rename ``run_application()`` to ``run_gui()``.
* Read the configuration file in **ts_config_mttcs**.
* Add the **signals.py**, **config.py**, and **status.py**.

### CODE REVIEW

File: doc/developer-guide/developer-guide.rst [hunk 1]
```diff
@@ -32,6 +33,39 @@ rotgui
...
+* **Config** is a data class that has the configuration details in the rotator control system.
+* **Status** is a data class that has the current controller status.
+
+The model–view–controller (MVC) architecture is used in this module.
+In this design, the view always shows the data sent from the model.
+This helps to minimize the business logic in view and makes the tests easier.
+
+The `Qt signal <https://doc.qt.io/qt-6/signalsandslots.html>`_ is used to do the data exchange.
+The `emit()` and `connect()` in the class diagrams mean the class **emits** a specific signal and **connects** it to a specific callback function.
+Signals are holded and emitted from the **Model** to simplify the management of signals.
```
Comment (couger01) [inline doc/developer-guide/developer-guide.rst]:
```suggestion
Signals are held and emitted from the **Model** to simplify the management of signals.
```

File: doc/developer-guide/developer-guide.rst [hunk 2]
```diff
@@ -32,6 +33,39 @@ rotgui
 * **MainWindow** is the main window of the application.
 * **Model** contains the main business logic in the application.
 * **ControlPanel** has the commands to control the controller.
+* **Config** is a data class that has the configuration details in the rotator control system.
+* **Status** is a data class that has the current controller status.
+
+The model–view–controller (MVC) architecture is used in this module.
+In this design, the view always shows the data sent from the model.
+This helps to minimize the business logic in view and makes the tests easier.
```
Comment (couger01) [inline doc/developer-guide/developer-guide.rst]:
```suggestion
This helps minimize the business logic in view and makes testing easier.
```

**Review by couger01 — APPROVED** [review]
LGTM but see comments for suggestions.

File: doc/developer-guide/developer-guide.rst (hunk 1 above)
Comment (teweitsai) [reply doc/developer-guide/developer-guide.rst]: Improved in the commit: [9135b21](https://github.com/lsst-ts/ts_rotgui/pull/3/commits/9135b21900f0f24ecdb4dc0b5f0b99fd9a2a9b39). Thanks!

File: doc/developer-guide/developer-guide.rst (hunk 2 above)
Comment (teweitsai) [reply doc/developer-guide/developer-guide.rst]: Improved in the commit: [9135b21](https://github.com/lsst-ts/ts_rotgui/pull/3/commits/9135b21900f0f24ecdb4dc0b5f0b99fd9a2a9b39). Thanks!

**Pull request merged**